vault/logical_system.go

```diff
@@ -1976,7 +2006,56 @@ func (b *SystemBackend) handleTuneWriteCommon(ctx context.Context, path string,
 		}
 	}
 
-	return nil, nil
+	var resp *logical.Response
+	if optionsRaw, ok := data.GetOk("options"); ok {
+		b.Core.logger.Info("core: mount tuning of options", "path", path)
+		options := optionsRaw.(map[string]interface{})
+		optionMap := make(map[string]string)
+		for k, v := range options {
+			vStr, ok := v.(string)
+			if !ok {
+				return logical.ErrorResponse("options must be string valued"),
+					logical.ErrInvalidRequest
+			}
+			optionMap[k] = vStr
+		}
+
+		b.Core.logger.Info("core: mount tuning of options", "path", path, "options", optionMap)
+		// Special case to make sure we can not disable versioning once it's
```

Comment:
We should also special case the "plugin_name" so that it cannot be changed via mount tune. We can support this behavior in the future, but I don't think that this is currently handled.

Reply:
Right now the plugin name will overwrite whatever was user supplied in the options map- not ideal but at least it cannot be changed. Eventually it would be good to move this out of the options map all together.

Reply:
Right, can we add the logic in here to ignore/not overwrite the value if it already exists?

Reply:
Nevermind, I think I am confusing `entry.Config` and `entry.Options`.